packages/in_app_purchase/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -92,7 +96,17 @@ - (void)paymentQueue:(SKPaymentQueue *)queue
 - (void)paymentQueue:(SKPaymentQueue *)queue
     removedTransactions:(NSArray<SKPaymentTransaction *> *)transactions {
   for (SKPaymentTransaction *transaction in transactions) {
-    [self.transactionsSetter removeObjectForKey:transaction.payment.productIdentifier];
+    NSString *productId = transaction.payment.productIdentifier;
+
+    if ([self.transactionsSetter objectForKey:productId] == nil) {
+      continue;
+    }
+
+    [self.transactionsSetter[productId] removeObjectsInArray:[self.transactionsSetter[productId] filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"transactionIdentifier == %@", transaction.transactionIdentifier]]];
```

Comment:
Consider making this a single line to help formatting.

Reply:
Done